pipeline_plugins/resource/hook/quicktime.py

```diff
@@ -119,27 +125,105 @@ def launch(self, event):
                     version.publish()
 
                 for c in version.getComponents():
-                    data.append({"label": c.getName(), "value": c.getId()})
+                    data.append({
+                        "label": c.getName(),
+                        "value": self.get_file_for_component(c)
+                    })
 
             # get all components on all valid versions
             if entityType == "task":
                 task = ftrack.Task(selection[0]["entityId"])
 
                 for asset in task.getAssets(assetTypes=["mov"]):
                     for version in asset.getVersions():
+                        version = str(version.getVersion()).zfill(3)
```

Comment:
There is an issue here where you are redefining ```version``` as a string and overwriting the Ftrack object.